Utils/Services/ImageFetcher.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  ImageFetcher.swift
+//  Utils
+//
+//  Created by Nahuel Gladstein on 6/13/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import UIKit
+import ReactiveSwift
+
+public protocol ImageFetcherType {
+    
+    func fetchImage(_ imageURL: URL) -> SignalProducer<UIImage, ImageFetcherError>
+    
+}
+
+public enum ImageFetcherError: Error {
+    case invalidImageFormat
+    case fetchError(Error)
+}
+
+public class ImageFetcher: ImageFetcherType {
+    
+    /**
+     Fetches an image asynchronously from a URL, returning a value or an error in the SignalProducer when done.
+     - parameter imageURL: the url of the image to fetch.
+     */
+    public func fetchImage(_ imageURL: URL) -> SignalProducer<UIImage, ImageFetcherError> {
+        return URLSession.shared
```

Comment:
I would add that to an `URLSession` extension, but I'm not sure how to show the interaction we are trying to add... Maybe in the documentation of the `fetchImage` function I can add a comment telling that you can use `URLSession.cacheSession` to use that cache policy